### PR TITLE
[BBPBGLIB-1122] Removing legacy RNGs, leaving only Random123

### PR DESCRIPTION
**Pull request by atemerev**
Removing legacy RNGs and the ability to configure RNGs. Random123 is chosen everywhere as the only option.

## Context

RNG types differing from Random123 are no longer used and removed from everywhere (including configuration).

## Scope

RNGSettings.hoc (removed RNGmode and all related constants), configuration.py, random.py, and corresponding usages.

## Testing

Write a test that raises exception in attempt to configure RNG mode.

## Review
* [X] PR description is complete
* [X] Coding style (imports, function length, New functions, classes or files) are good
* [ ] Unit/Scientific test added
* [ ] Updated Readme, in-code, developer documentation

### CODE REVIEW

File: neurodamus/data/hoc/RNGSettings.hoc
Comment (WeinaJi) [inline neurodamus/data/hoc/RNGSettings.hoc:1]: Can we replace the hoc template with the python class`RNGConfig` in `configuration.py`?

Comment (WeinaJi) [reply neurodamus/data/hoc/RNGSettings.hoc:1]: I see this hoc template is called by our synapse helper functions, such as https://github.com/BlueBrain/neurodamus-models/blob/main/common/hoc/AMPANMDAHelper.hoc . It is pity that the call is just to get a `SynapseSeed` which is set globally. We could pass the seed to the helper init function instead. Will leave the clean up later on.

File: neurodamus/core/random.py [hunk 1]
```diff
@@ -3,57 +3,36 @@
...
 #       Neuron objects are returned (using __new__)
 
 
 class RNG(object):
     def __new__(cls, *ids, **kw):
-        """Creates a default RNG, currently based on ACG"""
+        """Creates a default RNG (Random123) """
         seed = kw.get("seed")
-        return Neuron.h.Random() if seed is None else Neuron.h.Random(seed)
+        return Random123(ids[0], ids[1], ids[2], seed)
```
Comment (WeinaJi) [inline neurodamus/core/random.py:16]: A nitpick, shall we also assert len(ids) ?

Comment (atemerev) [reply neurodamus/core/random.py]: I think I had it the other way around, RNG() should create some default instance of a random generator (I am not sure this is a good idea, but it is used throughout the code, e.g. in `stimuli.py`). We can do it with seed=0 (deterministic) or seed=current time (stochastic), I think deterministic is OK for now.

File: neurodamus/core/random.py [hunk 2]
```diff
@@ -3,57 +3,36 @@
...
 
-# NOTE: These are pure wrappers, in the sense they dont create Python objects. Instead
+# NOTE: These are pure wrappers, in the sense they don't create Python objects. Instead
 #       Neuron objects are returned (using __new__)
 
 
 class RNG(object):
-    def __new__(cls, *ids, **kw):
+    def __new__(cls, **kw):
         """Creates a default RNG, currently based on ACG"""
```
Comment (WeinaJi) [inline neurodamus/core/random.py:14]: Just discussed with @alkino  and @pramodk , it looks good that we instantiate Random123 object with a (0,1,2, seed), the first 3 seeds will be changed later on during synapse creation by gid, syn_id, etc.

Another nitpick, could you please change the docstring to replace ACG with Random123 ?

File: neurodamus/data/hoc/RNGSettings.hoc [hunk 3]
```diff
@@ -17,15 +17,13 @@ globalSeed = 0
...
 public init, interpret, getRNGMode, getIonChannelSeed, getSynapseSeed, getStimulusSeed, getMinisSeed, getGlobalSeed
-public COMPATIBILITY, RANDOM123, UPMCELLRAN4
+public RANDOM123
 
 external rngMode, ionchannelSeed, synapseSeed, stimulusSeed, minisSeed, globalSeed, terminate
 
 proc init() {
     // consts for random number handling
-    COMPATIBILITY = 0
     RANDOM123 = 1
```
Comment (WeinaJi) [inline neurodamus/data/hoc/RNGSettings.hoc:26]: Can we also remove `rngMode` and `getRNGMode` ?  I see that you have removed them in`connection.py` and `stimulus_manager.py`.

Comment (atemerev) [reply neurodamus/data/hoc/RNGSettings.hoc:26]: OK, apparently, their usages were removed in neurodamus-models, so I can do that.

**Review by WeinaJi — APPROVED** [review]

**Pull request merged**